### PR TITLE
add backend ID to case and form ES indexes

### PR DESCRIPTION
**Pull request by snopoke**
@calellowitz having this is useful for some stats

### CODE REVIEW

**Comment by calellowitz** [comment]
test fail looks real
```
======================================================================
FAIL: test_migrate_ledgers (corehq.apps.couch_sql_migration.tests.test_migration.LedgerMigrationTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/couch_sql_migration/tests/test_migration.py", line 344, in test_migrate_ledgers
    self._compare_diffs([])
  File "/mnt/commcare-hq-ro/corehq/apps/couch_sql_migration/tests/test_migration.py", line 51, in _compare_diffs
    self.assertEqual(expected, json_diffs)
AssertionError: Lists differ: [] != [(u'XFormInstance', FormJsonDi...
Second list contains 5 additional elements.
First extra element 0:
(u'XFormInstance', FormJsonDiff(diff_type=u'missing', path=[u'backend_id'], old_value=Ellipsis, new_value=u'sql'))
- []
+ [(u'XFormInstance',
+   FormJsonDiff(diff_type=u'missing', path=[u'backend_id'], old_value=Ellipsis, new_value=u'sql')),
+  (u'XFormInstance',
+   FormJsonDiff(diff_type=u'missing', path=[u'backend_id'], old_value=Ellipsis, new_value=u'sql')),
+  (u'XFormInstance',
+   FormJsonDiff(diff_type=u'missing', path=[u'backend_id'], old_value=Ellipsis, new_value=u'sql')),
+  (u'XFormInstance',
+   FormJsonDiff(diff_type=u'missing', path=[u'backend_id'], old_value=Ellipsis, new_value=u'sql')),
+  (u'CommCareCase',
+   FormJsonDiff(diff_type=u'missing', path=[u'backend_id'], old_value=Ellipsis, new_value=u'sql'))]
```

**Comment by czue** [comment]
:+1:

File: corehq/pillows/mappings/case_mapping.py [hunk 1]
```diff
@@ -98,6 +98,7 @@
                       'type': 'string'},
         'contact_phone_number': {'index': 'not_analyzed',
                                  'type': 'string'},
+        'backend_id': {'type': 'string', 'index': 'not_analyzed'},
```
Comment (emord) [inline corehq/pillows/mappings/case_mapping.py]: Should you update the index names as well?

File: corehq/ex-submodules/pillowtop/management/commands/update_es_mapping.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+from corehq.elastic import get_es_new
+from corehq.pillows.utils import get_all_expected_es_indices
+
+
+class Command(BaseCommand):
+    help = "Update an existing ES mapping. If there are conflicting changes this command will fail."
+    args = "[INDEX NAME or ALIAS]"
+
+    def handle(self, *args, **options):
+        if len(args) != 1: raise CommandError("Please specify the index name to update")
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/management/commands/update_es_mapping.py]: E701 multiple statements on one line (colon)

File: corehq/ex-submodules/pillowtop/management/commands/update_es_mapping.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        indexes = [index for index in es_indices if index_name == index.alias or index_name == index.index]
+
+        if not indexes:
+            raise CommandError("No matching index found: {}".format(index_name))
+        index_info = indexes[0]
+        es = get_es_new()
+        if _confirm("Confirm that you want to update the mapping for '{}'".format(index_info.index)):
+            mapping = copy(index_info.mapping)
+            mapping['_meta']['created'] = datetime.utcnow().isoformat()
+            mapping_res = es.indices.put_mapping(index_info.type, {index_info.type: mapping}, index=index_info.index)
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/management/commands/update_es_mapping.py]: E501 line too long (117 > 115 characters)

File: corehq/pillows/mappings/case_mapping.py (hunk 1 above)
Comment (snopoke) [inline corehq/pillows/mappings/case_mapping.py]: I'm just going to update the mappings in place and then reindex the sql models

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**